File: Chapitre_4/programme_fils_exo4_4.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#define TAILLE 256

int main()
{
    int tube_0,tube_1;
    char *tube;
    char message[256];
    tube=getenv("DESC_0");
    tube_0=atoi(tube);
    tube=getenv("DESC_1");
    tube_1=atoi(tube);
    close (tube_1);
    read(tube_0,message,TAILLE);
    printf("Message : %s\n",message);
    return(0);
}

File: Chapitre_4/programme_fils_exo4_4.py
import os 
import sys

if len(sys.argv) >1 :
    tube=os.environ.get("DESC_0")
    tube_0=int (tube)
    tube=os.environ.get("DESC_1")
    tube_1=int (tube)
    os.close(tube_1)
    mess=os.read(tube_0,1024)
    os.close(tube_0)
    message=str(mess.decode())
    print("Message : {}".format(message))
